src/plugins/data_views/public/data_views/data_views_api_client.ts

```diff
@@ -104,7 +104,8 @@ export class DataViewsApiClient implements IDataViewsApiClient {
         allow_no_index: allowNoIndex,
         include_unmapped: includeUnmapped,
         fields,
-        allow_hidden: allowHidden,
+        // default to undefined to keep value out of URL params and improve caching
+        allow_hidden: allowHidden || undefined,
```

Comment:
This approach is much nicer than addressing each caller separately 👍 no need to try to remember it in future changes.